sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/optimizer/MergeScalarSubqueries.scala

```diff
@@ -353,9 +353,20 @@ object MergeScalarSubqueries extends Rule[LogicalPlan] {
         case a: AggregateExpression => a
       })
     }
+    val groupByExpressionSeq = Seq(newPlan, cachedPlan).map { plan =>
+      plan.groupingExpressions.flatMap(_.collect {
```

Comment:
I think it should be just `plan.groupingExpressions`, e.g. `GROUO BY func(a)` can be supported by hash aggregate if `func(a)` returns normal string, no matter if `a` is string with collation or not.

Reply:
Ups, my bad. Good catch! Let me see if I can add a test that verifies this logic.

Reply:
Added test please take a look.

I also added string with collation to `dsl/package` since that was a natural way to enable the optimizer tests and i guess that we need this anyhow, although this is not strictly related to aggs.